### PR TITLE
feat(frontend): handle TLM reconnection in case of error and not received keep-alives

### PR DESCRIPTION
**Pull request by grignards**
See #3203.

Display a message to the user if connection is lost for more than 4
seconds

*Please indicate here a summary of the pull request content. If a related issue exists, please indicate issue reference here*

## Checkpoints

The goal of this section is to help code reviewers and the merge request author to do the required checks on this pull request. Please don't edit this list.
If one or more of these checkpoints are out of scope, please write below why they are.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, then there are associated issues and issue is referenced in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present
- [x] Automated tests have been written (covering the feature or non regression), or *at least* an issue has been created to implement the test

**For developers**

- [x] Manual tests done to ensure stability on whole application layers, issue expectation follow
- [x] The original issue has been updated with a short summary of the implemented solution

**For the quality team**

- [x] Tested by the quality team

### CODE REVIEW

File: frontend/test/src/LiveMessageManager.spec.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    it('should set manager status to CLOSED', () => {
+      const { manager } = openedManager()
+      managers.push(manager)
+      manager.closeLiveMessageConnection()
+      expect(manager.status).to.be.equal(LIVE_MESSAGE_STATUS.CLOSED)
+    })
+  })
+
+  describe('the heartbeat timer', () => {
+    it('should restart connection when the heartbeat event is not received in time', async () => {
```
Comment (inkhey) [inline frontend/test/src/LiveMessageManager.spec.js:59]: Just a question, Do you expect heartbeat event at a regular time ?
If so, you should probably modify a bit https://github.com/tracim/tracim/blob/1e9d9c30fd4e7e2de62888b0b81d83de358cc005/backend/tracim_backend/views/core_api/user_controller.py#L626
with "interval"
see https://pushpin.org/docs/advanced/#keep-alives

Comment (grignards) [reply frontend/test/src/LiveMessageManager.spec.js:59]: Ah, yes you're right I saw this… I can also reset the frontend timer when a tlm is received :thinking:

File: frontend/src/component/FlashMessage.styl [hunk 2]
```diff
@@ -38,6 +38,9 @@
         &__paragraph
           padding-right 20px
 
+#connection_error
```
Comment (Skylsmoi) [inline frontend/src/component/FlashMessage.styl:41]: avoid using html ids for styling, prefer class usage
The idea behind it is to keep specificity low and manageable

File: frontend/src/container/Tracim.jsx [hunk 3]
```diff
@@ -247,6 +282,18 @@ export class Tracim extends React.Component {
...
+            id='connection_error'
+            flashMessage={
+              [{
+                message: props.t('Tracim has a connection problem, please wait or reload the page if the problem persists'),
+                type: 'danger'
+              }]
+            }
+            showCloseButton={false}
+            t={props.t}
+          />}
```
Comment (Skylsmoi) [inline frontend/src/container/Tracim.jsx:296]: to have the closing bracket at the same column as the opening one while having standardjs ok with it is by using parenthesis like so
```suggestion
        {state.displayConnectionError && (
          <FlashMessage
            id='connection_error'
            flashMessage={
              [{
                message: props.t('Tracim has a connection problem, please wait or reload the page if the problem persists'),
                type: 'danger'
              }]
            }
            showCloseButton={false}
            t={props.t}
          />
        )}
```

File: frontend/src/reducer/tlmManager.js [hunk 4]
```diff
@@ -1,9 +1,11 @@
-import { SET, TLM_MANAGER } from '../action-creator.sync.js'
+import { SET, TLM_MANAGER, TLM_MANAGER_STATUS } from '../action-creator.sync.js'
 
-export default function tlmManager (state = null, action) {
+export default function tlm (state = {}, action) {
```
Comment (Skylsmoi) [inline frontend/src/reducer/tlmManager.js:3]: If you rename the reducer, also rename its filename please

Comment (Skylsmoi) [reply frontend/src/reducer/tlmManager.js:3]: Also, now you have multiple properties, its better to add them to its default value
```jsx
const defaultTlm = {
  manager: null,
  status: LIVE_MESSAGE_STATUS.CLOSED
}
export default function tlm (state = defaultTlm, action) {
```

**Pull request merged**